src/util/gobConfig.ml

```diff
@@ -304,22 +336,25 @@ struct
   let get_string_list = List.map Yojson.Safe.Util.to_string % get_list
 
   (** Helper functions for writing values. *)
-  let set_path_string st v =
-    set_value v json_conf (parse_path st)
+  (* let set_path_string st v =
+    set_value v json_conf (parse_path st) *)
 
-  (** Helper functions for writing values. Handels the tracing. *)
-  let set_path_string_trace st v =
+  (** Sets a value, preventing changes when the configuration is immutable and invalidating the cache. *)
+  let set_value v o pth =
+    if is_immutable () then raise (Immutable pth);
     drop_memo ();
-    if tracing then trace "conf" "Setting '%s' to %a.\n" st GobYojson.pretty v;
-    set_path_string st v
+    unsafe_set_value v o pth
+
+  (** Helper function for writing values. Handles the tracing. *)
+  let set_path_string ?(show_trace = true) st v =
+    if show_trace && tracing then trace "conf" "Setting '%s' to %a.\n" st GobYojson.pretty v;
+    set_value v json_conf (parse_path st)
 
   (** Convenience functions for writing values. *)
-  let set_int    st i = set_path_string_trace st (`Int i)
-  let set_bool   st i = set_path_string_trace st (`Bool i)
-  let set_string st i = set_path_string_trace st (`String i)
-  let set_list   st l =
-    drop_memo ();
-    set_value (`List l) json_conf (parse_path st)
+  let set_int    st i = set_path_string st (`Int i)
+  let set_bool   st i = set_path_string st (`Bool i)
+  let set_string st i = set_path_string st (`String i)
+  let set_list   st l = set_path_string ~show_trace:false st (`List l)
```

Comment:
I think it's just a historical accident that `set_list` didn't do the tracing. You can just remove this argument and make tracing unconditional.